python-language-server "MSBUILD : error MSB1001: Unknown switch."

### Describe the bug
`python-language-server` no longer builds

### Steps To Reproduce
Steps to reproduce the behavior:
1. Be on `nixos-unstable`
1. Include `python-language-server` in `environment.systemPackages`
1. `sudo nixos-rebuild switch --upgrade`

### Expected behavior
Expect `python-language-server` to build without errors.

### Screenshots
Not applicable.

### Additional context

```console
✦ ❯ sudo nix-channel --list
nixos https://nixos.org/channels/

✦ ❯  sudo nixos-rebuild switch --upgrade
unpacking channels...
building Nix...
building the system configuration...
these 12 derivations will be built:
  /nix/store/fbnrkijbh9zc9md1z9fvjqx3g71f2wbj-python-language-server-2022-02-18.drv
  /nix/store/h6is7agjyv4yrqls9axf7nd6pvf2mvlq-system-path.drv
  /nix/store/5092gz11s6cl1pznd5sd41rz5lfhhfbq-dbus-1.drv
  /nix/store/c094w8hz52wb7g5m3ics22cclgfqfpl0-unit-polkit.service.drv
  /nix/store/fj2i1rmrdw8dgvwhi6fsi6skim9nr6h7-unit-dbus.service.drv
  /nix/store/iqf48pvg688fd2r3ksbb34vfy83khl2x-unit-systemd-fsck-.service.drv
  /nix/store/sh3q0vihq9r1j62hi140rxrb0av3232b-unit-accounts-daemon.service.drv
  /nix/store/6vykcby1xc0vpgki62i1vasp9pnh2hk8-system-units.drv
  /nix/store/p8r3rav108zrl97x1wyriddfjsk2pjdz-unit-dbus.service.drv
  /nix/store/nimjlk4d2v31yrrjcf8shc7bcjsdkn4a-user-units.drv
  /nix/store/a7g2wm3ghjknxkk116xjd68fj7r53iab-etc.drv
  /nix/store/l4jyvs85zxnf5xam0ybk55rx8cjxg1v5-nixos-system-wombat11k-22.11pre395754.2a93ea177c3.drv
building '/nix/store/fbnrkijbh9zc9md1z9fvjqx3g71f2wbj-python-language-server-2022-02-18.drv'...
unpacking sources
unpacking source archive /nix/store/gf7fp5k5q1hn09mbrm5sw7a1bcj20n2f-source
source root is source
patching sources
configuring
Executing dotnetConfigureHook
  Determining projects to restore...
  Restored /build/source/src/Analysis/Core/Impl/Microsoft.Python.Analysis.Core.csproj (in 207 ms).
  Restored /build/source/src/LanguageServer/Impl/Microsoft.Python.LanguageServer.csproj (in 3.17 sec).
  Restored /build/source/src/Parsing/Impl/Microsoft.Python.Parsing.csproj (in 10 ms).
  Restored /build/source/src/Core/Impl/Microsoft.Python.Core.csproj (in 8 ms).
  Restored /build/source/src/Analysis/Ast/Impl/Microsoft.Python.Analysis.csproj (in 9 ms).
Finished dotnetConfigureHook
building
Executing dotnetBuildHook
Microsoft (R) Build Engine version 16.7.2+b60ddb6f4 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1001: Unknown switch.
Switch: --no-self-contained

For switch syntax, type "MSBuild -help"
error: builder for '/nix/store/fbnrkijbh9zc9md1z9fvjqx3g71f2wbj-python-language-server-2022-02-18.drv' failed with exit code 1
error: 1 dependencies of derivation '/nix/store/h6is7agjyv4yrqls9axf7nd6pvf2mvlq-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/l4jyvs85zxnf5xam0ybk55rx8cjxg1v5-nixos-system-wombat11k-22.11pre395754.2a93ea177c3.drv' failed to build
```

### Notify maintainers
@thomasjm
@mdarocha 

### Metadata

```console
✦ ❯ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.06 MiB download, 0.30 MiB unpacked):
  /nix/store/a9f4h2yjhdlk7xj3wj3p7b5r6s659qfl-bash-interactive-5.1-p16-dev
copying path '/nix/store/a9f4h2yjhdlk7xj3wj3p7b5r6s659qfl-bash-interactive-5.1-p16-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.56, NixOS, 22.11 (Raccoon), 22.11pre395754.2a93ea177c3`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.10.3`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos
 ```

--- a/pkgs/build-support/dotnet/build-dotnet-module/hooks/dotnet-build-hook.sh
+++ b/pkgs/build-support/dotnet/build-dotnet-module/hooks/dotnet-build-hook.sh
@@ -15,9 +15,9 @@ dotnetBuildHook() {
     fi
 
     if [ "${selfContainedBuild-}" ]; then
-        dotnetBuildFlags+=("--self-contained")
+        dotnetBuildFlags+=("-p:SelfContained=true")
     else
-        dotnetBuildFlags+=("--no-self-contained")
+        dotnetBuildFlags+=("-p:SelfContained=false")
     fi
 
     if [ "${version-}" ]; then

